### DISCUSSION TITLE
Why get two J (torsional constant ) values with and without material properties

### DISCUSSION THREAD

**Discussion by nira123** [robbievanleeuwen/section-properties#248] [Q&A] [answered]
Please bare with me if I am wrong. I though J (torsional constant ) is dependent on the geometric shape but not on the materiel properties. 
Commenting geometry in following code with and without material, I get two values for torsional constant (j) 
Could someone please let me know if i have done anything wrong or any reference where J is dependent on material properties.  

```python
from sectionproperties.pre.geometry import Geometry
from sectionproperties.analysis.section import Section
from sectionproperties.pre.pre import Material
from shapely.geometry import Polygon, LineString, Point
from shapely.ops import unary_union

precast = Material(
    name="65 MPa",
    elastic_modulus=37.4e3,
    poissons_ratio=0.2,
    yield_strength=65,
    density=2.4e-6,
    color="grey",
)

poly_2=[(-1000, 1150),
 (1000, 1150),
 (1000, 1300),
 (-1000, 1300)]


p2 = Polygon(poly_2)
#geometry = Geometry(p2, material=precast)
geometry = Geometry(p2)
geometry.create_mesh(mesh_sizes=[1000])

section = Section(geometry)
section.calculate_geometric_properties()
section.calculate_warping_properties()
section.get_j()
```

**Comment by robbievanleeuwen** [comment, marked as answer]
Hi @nira123, please refer to the documentation, [How Material Properties Affect Results
](https://sectionproperties.readthedocs.io/en/latest/rst/post.html#how-material-properties-affect-results)

In summary, if you provide material properties, ``get_j()`` returns the modulus weighted ``E.J``.

**Reply by nira123** [reply]
Hi Robbie,
Many thanks for your response. That's good you allow EJ for torsional stiffness where  material is defined specially when dealing with composite sections.  
Sometimes we need to get j and wrapping properties together, one may misinterpret that as purely J .   I think it is good to mention somewhere with pre-defined material, get_j() return EJ . 
Thanks for developing this awesome module.

**Reply by robbievanleeuwen** [reply]
Thanks @nira123.

I am happy with how this is mentioned in the documentation I think it is very clear. Under [Defining Material Properties](https://sectionproperties.readthedocs.io/en/latest/rst/geom_mesh.html#defining-material-properties) there is a warning that directs you to [How Material Properties Affect Results](https://sectionproperties.readthedocs.io/en/latest/rst/post.html#label-material-results). In this section it directly says:

> Torsion constant ``get_j()`` - returns elastic modulus weighted torsion constant ``E.J``

**Reply by nira123** [reply]
Thanks a lot Robbie. Aha sorry didn't notice that.
All good, Thanks for your prompt response.
